File: 4-kyu/Valid Braces/ValidBraces.cpp
// Kata located at: https://www.codewars.com/kata/5277c8a221e209d3f6000b56

bool valid_braces(std::string braces) 
{
  std::stack<int> s;
  for(int i = 0; i < braces.length(); ++i)	{
  
  	char c = braces[i];
    if(c == '(' || c == '[' || c == '{')	s.push(c);
    else if(c == ')')	{
    
    	if(s.empty())	return false;
      else if(s.top() == '(') s.pop();
      else return false;
    
    }	else if(c == ']')	{
    
    	if(s.empty())	return false;
      else if(s.top() == '[')	s.pop();
      else return false;
    
    }	else if(c == '}')	{
    
    	if(s.empty())	return false;
      else if(s.top() == '{')	s.pop();
      else return false;
    
    }
  
  }
  
  return s.empty();
  
}

File: 4-kyu/Valid Braces/ValidBraces.java
// Kata found at: https://www.codewars.com/kata/5277c8a221e209d3f6000b56

import java.util.Stack;
public class BraceChecker {

  public boolean isValid(String braces) {
  // ()[]{}
  
  	Stack<Character> stack = new Stack<Character>();
  	for(int i = 0; i < braces.length(); ++i)	{
    
    	char c = braces.charAt(i);
      if(c == '(' || c == '[' || c == '{')	stack.push(c);
      else if(c == ')')	{
      	
        if(stack.isEmpty())	return false;
        else if(stack.peek() == '(')	stack.pop();
        else return false;
      
      }	else if(c == ']')	{
      
      	if(stack.isEmpty())	return false;
        else if(stack.peek() == '[')	stack.pop();
        else return false;
      
      }	else if(c == '}')	{
      
      	if(stack.isEmpty())	return false;
        else if(stack.peek() == '{')	stack.pop();
        else return false;
      
      }
      
    }
    
    return stack.isEmpty();
    
  }

}
